tests/integration/orcid/test_orcid_putcode_getter.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+import pytest
+
+from inspirehep.modules.orcid import exceptions
+from inspirehep.modules.orcid.putcode_getter import OrcidPutcodeGetter
+
+from utils import override_config
+
+
+@pytest.mark.usefixtures('isolated_app')
+class TestOrcidPutcodeGetter(object):
+    def setup(self):
+        self.orcid = '0000-0002-6665-4934'  # ATLAS author.
+        from flask import current_app  # Note: isolated_app not available in setup().
+        # Pick the token from local inspirehep.cfg first.
+        self.oauth_token = current_app.config['ORCID_APP_CREDENTIALS'].get('oauth_tokens', {}).get(self.orcid, 'mytoken')
```

Comment:
Why do we care about getting the token from the application config if it's the test run -- i.e. the token _is supposed to be_ garbage?

Reply:
If you keep actual credentials (working against ORCID.org PROD) in your local `inspirehep.cfg` config file (ignored by git), then you will be able to record new cassettes/episodes effortless when you need - thanks to this line.
Note that VCR is configured to ignore the auth headers, so all secrets are kept safe.